src/types.ts

```diff
@@ -347,7 +347,7 @@ export type BlockValueProp<T> = Extract<BlockValueType, { type: T }>;
 export interface CustomBlockComponentProps<T extends BlockValueTypeKeys> {
   renderComponent: () => JSX.Element | null;
   blockMap: BlockMapType;
-  blockValue: T extends BlockValueType ? BlockValueProp<T> : BaseValueType;
```

Comment:
BlockValueTypeKeys is union of string, but BlockValueType not, T would never match `BlockValueTypeKeys & BlockValueType`